caikit_nlp/modules/text_classification/sequence_classification.py

```diff
@@ -0,0 +1,235 @@
+                Path to the model to be loaded.
+
+        Returns:
+            SequenceClassification
+                Instance of this class built from the on disk model.
+        """
+        tokenizer = AutoTokenizer.from_pretrained(model_path)
+        model = AutoModelForSequenceClassification.from_pretrained(model_path)
+        device = SequenceClassification._get_device()
+        return cls(
+            tokenizer=tokenizer,
+            model=model,
+            device=device,
+        )
+
+    ################################## Private Functions #########################################
+
+    def _get_scores(self, text: Union[str, List[str]]):
+        """Run tokenizer and model to get scores on text(s)
+
+        Args:
+            text: Union[str, List[str]]
+                Input string(s) to be used
+
+        Returns:
+            scores_dict
+                Dict with key label, and values as the array of scores,
+                each corresponding to text(s)
+        """
+        # Apply tokenizer
+        tokenized_text = self.tokenizer(
+            text,
+            padding="max_length",
+            max_length=self.tokenizer.model_max_length,
+            truncation=True,
+            return_tensors="pt",  # PyTorch
+        )
+        # NOTE: no simple/efficient way to detect whether truncation
+        # happened since padding also occurs, and this can be applied on
+        # a batch of strings.
+        if self.device == "cuda":
+            tokenized_text = tokenized_text.to("cuda")
+        with torch.no_grad():
+            logits = self.model(**tokenized_text).logits
+
+        if not self.model.config.id2label:
+            log.warning(
+                "<NLP31047577W>",
+                "No id2label provided in model config. Defaulting to numeric labels",
+            )
+
+        softmax = torch.nn.Softmax(dim=1)
+        raw_scores = softmax(logits)
+        if self.device == "cuda":
+            scores = raw_scores.cpu().numpy()
+        else:
+            scores = raw_scores.numpy()
+
+        scores_dict = {}
+        num_labels = self.model.num_labels
+        for label_idx in range(num_labels):
+            if self.model.config.id2label:
+                label = self.model.config.id2label[label_idx]
+            else:
+                label = label_idx
+            label_scores = scores[:, label_idx]
+            scores_dict[label] = label_scores
+        return scores_dict
+
+    @staticmethod
+    def _process_predictions(scores_dict: Dict, text_idx: int) -> ClassificationResult:
+        """Process dictionary of label: scores to ClassificationResult
+
+        Args:
+            scores_dict: Dict
+                Dict with key label, and values as the array of scores,
+                each corresponding to text(s)
+            text_idx: int
+                Integer index of text in batch
+
+        Returns:
+            ClassificationResult
+        """
+        classification_list = []
```

Comment:
should we add a typecheck here to make sure scores_dict is actually a dictionary?

Reply:
we control the dict creation for use in this private function so I can/will add a check but not sure if it's completely necessary